### PR TITLE
Re-add regularization section, with weight distributions.

### PR DESCRIPTION
**Pull request by MarkDaoust**
Staging:

https://colab.research.google.com/github/markdaoust/models/blob/add-notebook/samples/core/tutorials/estimators/linear.ipynb

### CODE REVIEW

File: samples/core/tutorials/estimators/linear.ipynb [hunk 1]
```diff
@@ -17,7 +23,8 @@
       "display_name": "Python 3",
       "language": "python",
       "name": "python3"
-    }
+    },
+    "accelerator": "GPU"
```
Comment (lamberta) [inline samples/core/tutorials/estimators/linear.ipynb]: Did you mean to turn this on?

File: samples/core/tutorials/estimators/linear.ipynb [hunk 2]
```diff
@@ -399,7 +417,7 @@
       },
       "cell_type": "markdown",
       "source": [
-        "[pandas](https://pandas.pydata.org/) provides some convenient utilities for data analysis. Here's a list of columns available in the Census Income dataset:"
+        "[Pandas](https://pandas.pydata.org/) provides some convenient utilities for data analysis. Here's a list of columns available in the Census Income dataset:"
```
Comment (lamberta) [inline samples/core/tutorials/estimators/linear.ipynb]: Its name is actually "pandas" (lowercase).

File: samples/core/tutorials/estimators/linear.ipynb [hunk 3]
```diff
@@ -1399,6 +1422,211 @@
...
+    {
+      "metadata": {
+        "id": "oyKy1lM_3gkL",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "## Adding Regularization to Prevent Overfitting\n",
+        "\n",
+        "Regularization is a technique used to avoid **overfitting**. Overfitting happens\n",
```
Comment (lamberta) [inline samples/core/tutorials/estimators/linear.ipynb]: Copy edit:

Regularization is a technique used to avoid overfitting. Overfitting happens when a model performs well on the data it is trained on, but worse on test data that the model has not seen before. Overfitting can occur when a model is excessively complex, such as having too many parameters relative to the number of observed training data. Regularization allows you to control the model's complexity and make the model more generalizable to unseen data.

You can add L1 and L2 regularizations to the model with the following code:

File: samples/core/tutorials/estimators/linear.ipynb [hunk 4]
```diff
@@ -1399,6 +1422,211 @@
...
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "Lp1Rfy_k4e7w",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "These regularized models don't perform very differently base model. Let look ar the models' weight distributions to better see the effect of the regularization:"
```
Comment (lamberta) [inline samples/core/tutorials/estimators/linear.ipynb]: Copy/typo edit:

These regularized models don't perform much better than the base model. Let's look at the model's weight distributions to better see the effect of the regularization:

File: samples/core/tutorials/estimators/linear.ipynb [hunk 5]
```diff
@@ -1399,6 +1422,211 @@
...
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "GskJmtfmL0p-",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "The models have many zero-vlaued weights  caused by unused hash bins (There are many more hash bins than categories in some columns). We will mask these weights when viewing the weight distributions:"
```
Comment (lamberta) [inline samples/core/tutorials/estimators/linear.ipynb]: Copy:

The models have many zero-valued weights caused by unused hash bins (there are many more hash bins than categories in some columns). We can mask these weights when viewing the weight distributions:

File: samples/core/tutorials/estimators/linear.ipynb [hunk 6]
```diff
@@ -1399,6 +1422,211 @@
...
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "Mv6knhFa5-iJ",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "Both types of regularization squeeze the distribution of weights towards zero. L2 regularization has a greater effect in the tails of the distribution eliminating extreme weights. L1 regularization produces more exactly-zero values (In this case it sets ~200 to zero)."
```
Comment (lamberta) [inline samples/core/tutorials/estimators/linear.ipynb]: Copy edit:

Both types of regularization squeeze the distribution of weights towards zero. L2 regularization has a greater effect in the tails of the distribution eliminating extreme weights. L1 regularization produces more exactly-zero values, in this case it sets ~200 to zero.

File: samples/core/tutorials/estimators/linear.ipynb (hunk 1 above)
Comment (MarkDaoust) [reply samples/core/tutorials/estimators/linear.ipynb]: Not getting a performance improvement. Removed.

File: samples/core/tutorials/estimators/linear.ipynb (hunk 2 above)
Comment (MarkDaoust) [reply samples/core/tutorials/estimators/linear.ipynb]: !

Done.

File: samples/core/tutorials/estimators/linear.ipynb (hunk 3 above)
Comment (MarkDaoust) [reply samples/core/tutorials/estimators/linear.ipynb]: done

File: samples/core/tutorials/estimators/linear.ipynb (hunk 4 above)
Comment (MarkDaoust) [reply samples/core/tutorials/estimators/linear.ipynb]: Done

File: samples/core/tutorials/estimators/linear.ipynb (hunk 5 above)
Comment (MarkDaoust) [reply samples/core/tutorials/estimators/linear.ipynb]: Done

File: samples/core/tutorials/estimators/linear.ipynb (hunk 6 above)
Comment (MarkDaoust) [reply samples/core/tutorials/estimators/linear.ipynb]: Done
